forward/forwarder.go

```diff
@@ -109,15 +115,64 @@ func NewForwarder(s Sender, ch shared.SubChannel, logger log.Logger) *Forwarder
 	}
 }
 
+func (f *Forwarder) emit(event interface{}) {
+	for _, listener := range f.listeners {
+		go listener.HandleEvent(event)
```

Comment:
I'm not sure what the implications for concurrency complexity are for a line like this. Is this safe? What are the disadvantages to do it procedural?